R/bsts_model.R

```diff
@@ -0,0 +1,52 @@
+#' BSTS model wrapper
+#'
+#' @param y Numeric vector of time points to forecast
+#' @param samples Numeric, number of samples to take.
+#' @param model A `bsts` model object wrapped in a function with an `ss` and `y` argument.
+#' @param horizon Numeric, the time horizon over which to predict.
+#' @return A dataframe of predictions (with columns representing the time horizon and rows representing samples).
+#' @export
+#' @importFrom bsts bsts predict.bsts
+#' @examples
+#'
+#' ## Used on its own
+#' bsts_model(y = EpiSoon::example_obs_rts[1:10, ]$rt,
+#'            model = function(ss, y){
+#'            bsts::AddAr(ss, y = y, lags = 2)},
+#'            samples = 10, horizon = 7)
+#'
+#' ## Used for forecasting
+#'  forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'                     model = function(...){EpiSoon::bsts_model(model =
```

Comment:
the nested function looks like quite an ingenious hack, but somehow still less then ideal. 
Also at line 28 you lose the flexibility to add something to ss, if you force it to pass in list()
don't think this is worth working on right now, just wanted to flag it.

Reply:
Keen to see an alternative implementation?

Reply:
Perhaps open an issue outlining the current limitations and we can review if a solution is immediatly obvious

Reply:
okk :)